pkg/apiserver/rest/webservice/application.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webservice
+
+import (
+	restfulspec "github.com/emicklei/go-restful-openapi/v2"
+	restful "github.com/emicklei/go-restful/v3"
+
+	apis "github.com/oam-dev/kubevela/pkg/apiserver/rest/apis/v1"
+)
+
+type applicationWebService struct {
+}
+
+func (c *applicationWebService) GetWebService() *restful.WebService {
+	ws := new(restful.WebService)
+	ws.Path(versionPrifix+"/applications").
+		Consumes(restful.MIME_XML, restful.MIME_JSON).
+		Produces(restful.MIME_JSON, restful.MIME_XML).
+		Doc("api for application manage")
+
+	tags := []string{"application"}
+
+	ws.Route(ws.GET("/").To(noop).
+		Doc("list all applications").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(ws.QueryParameter("query", "Fuzzy search based on name or description").DataType("string")).
+		Param(ws.QueryParameter("namespace", "Namespace-based search").DataType("string")).
+		Param(ws.QueryParameter("cluster", "Cluster-based search").DataType("string")).
+		Writes(apis.ListApplicationResponse{}))
+
+	ws.Route(ws.POST("/").To(noop).
+		Doc("create one application").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Reads(apis.CreateApplicationRequest{}).
+		Writes(apis.ApplicationBase{}))
+
+	ws.Route(ws.DELETE("/{name}").To(noop).
+		Doc("delete one application").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(ws.PathParameter("name", "identifier of the application").DataType("string")).
+		Writes(apis.ApplicationBase{}))
+
+	ws.Route(ws.GET("/{name}").To(noop).
+		Doc("detail one application").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(ws.PathParameter("name", "identifier of the application").DataType("string")).
+		Writes(apis.DetailApplicationResponse{}))
+
+	ws.Route(ws.POST("/{name}/template").To(noop).
+		Doc("create one application template").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(ws.PathParameter("name", "identifier of the application").DataType("string")).
+		Reads(apis.CreateApplicationTemplateRequest{}).
+		Writes(apis.ApplicationTemplateBase{}))
+
+	ws.Route(ws.POST("/{name}/deploy").To(noop).
```

Comment:
`POST /application/_name/deploy` this is not restful API.

`POST /application/_name` this should be the deploy API in restful style.

In this case, if you want to record the application meta without deploy it, you should use some thing like:

```
POST /application/meta/_name
```